mariadb:latest image fails to build (MariaDB 11 version bump)

Originally reported by @quicksketch

### Current Behavior

3 days ago the `mariadb:latest` image bumped from MariaDB 10 to version 11 in this commit: https://github.com/TugboatQA/dockerfiles/commit/9e11822fb43f0a92bb5abe2e4b7597fd8fd70d19#diff-1f29ef5c43985fdd5f18d159a2a3e81f6492ec961a0ff430a7d78d2d40a522eb

Tugboat fails to build this image even with no configuration with the following errors in the log:

```
        /usr/local/bin/docker-entrypoint.sh: line 105: mysqld: command not found
2023-06-13 18:29:33+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.0.2+maria~ubu2204 started.
2023-06-13 18:29:33+00:00 [ERROR] [Entrypoint]: mariadbd failed while attempting to check config
        command was: mysqld --verbose --help
```

I think this is because MariaDB stopped bundling the backwards-compatible `mysql` symlinks in version 11. See https://mariadb.com/kb/en/mariadb-11-0-0-release-notes/ and in particular this issue: https://jira.mariadb.org/browse/MDEV-30203

So the `docker-entrypoint.sh` checking for `mysqld` will now fail, because that command is no longer bundled by default.

### Expected Behavior

Previews that use `mariadb:latest` or `mariadb:11` should build.

I'm not sure if this Tugboat-specific or an issue with MariaDB's own images.

### Steps To Reproduce

Build a Tugboat preview that has a service that uses MariaDB latest or 11:

```
services:
  php:
    image: tugboatqa/php:8.1-apache
    default: true
    depends: mariadb
  mariadb:
    image: tugboatqa/mariadb:latest
```

--- a/services/mariadb/run
+++ b/services/mariadb/run
@@ -1,2 +1,2 @@
 #!/bin/sh
-exec docker-entrypoint.sh mysqld
+exec docker-entrypoint.sh mariadbd

